src/plugins/vis_types/xy/public/config/get_config.ts

```diff
@@ -51,13 +52,17 @@ export function getConfig(
 
   const yAxes: Array<AxisConfig<ScaleContinuousType>> = [];
 
+  // avoid duplicates based on aggId
+  const aspectVisited = new Set();
```

Comment:
nit: it seems to me that it still does not work correctly enough. I have created 3 metrics that shares one Y-axis. I expect that there will be 1 element in the `yAxes` array

![image](https://user-images.githubusercontent.com/20072247/147937921-dcb5a410-090c-420d-8a09-ae53f3c92506.png)

This does not affect the UI, but since you have already touched this code, it might make sense to fix it.

Reply:
as a simple fix we can use added `Set` for storing id's.....

Reply:
Not sure this is really in scope of this PR. I think we could workout a better fix for that in a follow up in the 8.x line.

Reply:
Yes I agree with Marco, as this is going to be backported in 7.17, let's just fix the bug and not introduce more changes.